tests/cross/src/test/scala/tests/pc/CompletionMatchSuite.scala

```diff
@@ -242,4 +242,42 @@ class CompletionMatchSuite extends BaseCompletionSuite {
       |}""".stripMargin,
     filter = _.contains("exhaustive")
   )
+
+  checkEdit(
+    "exhaustive-upper-type-bounds",
+    """
+      |package example
+      |
+      |sealed trait TestA
+      |case object Foo extends TestA
+      |case object Bar extends TestA
+      |sealed trait TestB
+      |case object Baz extends TestB
+      |case object Goo extends TestB
+      |
+      |object Main {
+      |  def testExhaustive[T <: TestA with TestB](test: T): Boolean =
```

Comment:
Could we maybe add a test also with `def testExhaustive[T <: TestA](test: T): Boolean =`

To be sure it works without refinement too?

Reply:
Added